packages/next/src/build/webpack/loaders/next-flight-client-entry-loader.ts

```diff
@@ -15,19 +15,21 @@ export default function transformSource(this: any) {
   let { modules, server }: NextFlightClientEntryLoaderOptions =
     this.getOptions()
   const isServer = server === 'true'
+  const clientImports = JSON.parse(modules) as [string, string[]][]
 
-  if (!Array.isArray(modules)) {
-    modules = modules ? [modules] : []
-  }
-
-  const requests = modules as string[]
-  const code = requests
+  const code = clientImports
     // Filter out CSS files in the SSR compilation
-    .filter((request) => (isServer ? !regexCSS.test(request) : true))
-    .map(
-      (request) =>
-        `import(/* webpackMode: "eager" */ ${JSON.stringify(request)})`
-    )
+    .filter(([request]) => (isServer ? !regexCSS.test(request) : true))
+    .map(([request, imports]) => {
+      if (imports[0] === '*') {
+        return `import(/* webpackMode: "eager" */ ${JSON.stringify(request)})`
+      }
+
+      const uniqueImports = [...new Set(imports)]
+      return `import(/* webpackMode: "eager" *//* webpackExports: [${uniqueImports
```

Comment:
```suggestion
      return `import(/* webpackMode: "eager", webpackExports: [${uniqueImports
```

a comma would do it too